src/fsharp/vs/service.fs

```diff
@@ -1096,50 +1098,107 @@ type TypeCheckInfo
-              // This is probably old code from when we supported 'go to definition' generating IL metadata.
-              let item =
+          | Some (item :: _, _, _, _) ->
+              let getTypeVarNames (ilinfo: ILMethInfo) =
+                  let classTypeParams = ilinfo.DeclaringTyconRef.ILTyconRawMetadata.GenericParams |> List.map (fun paramDef -> paramDef.Name)
+                  let methodTypeParams = ilinfo.FormalMethodTypars |> List.map (fun typ -> typ.Name)
+                  classTypeParams @ methodTypeParams |> Array.ofList
+
+              let result =
                   match item.Item with
-                  | Item.MethodGroup (_, (ILMeth (_,ilinfo,_)) :: _, _) 
-                  | Item.CtorGroup (_, (ILMeth (_,ilinfo,_)) :: _) -> Item.Types ("", [ ilinfo.ApparentEnclosingType ])
-                  | Item.ILField (ILFieldInfo (typeInfo, _)) -> Item.Types ("", [ typeInfo.ToType ])
-                  | Item.ImplicitOp(_, {contents = Some(TraitConstraintSln.FSMethSln(_, vref, _))}) -> Item.Value(vref)
-                  | _                                         -> item.Item
-
-              let fail defaultReason = 
-                  match item with            
+                  | Item.CtorGroup (_, (ILMeth (_,ilinfo,_)) :: _) ->
+                      match ilinfo.MetadataScope with
+                      | ILScopeRef.Assembly assref ->
+                          let typeVarNames = getTypeVarNames ilinfo
+                          ParamTypeSymbol.tryOfILTypes typeVarNames ilinfo.ILMethodRef.ArgTypes
+                          |> Option.map (fun args ->
+                              let externalSym = ExternalSymbol.Constructor (ilinfo.ILMethodRef.EnclosingTypeRef.FullName, args)
+                              FSharpFindDeclResult.ExternalDecl (assref.Name, externalSym))
+                      | _ -> None
+
+                  | Item.MethodGroup (name, (ILMeth (_,ilinfo,_)) :: _, _) ->
+                      match ilinfo.MetadataScope with
+                      | ILScopeRef.Assembly assref ->
+                          let typeVarNames = getTypeVarNames ilinfo
+                          ParamTypeSymbol.tryOfILTypes typeVarNames ilinfo.ILMethodRef.ArgTypes
+                          |> Option.map (fun args ->
+                              let externalSym = ExternalSymbol.Method (ilinfo.ILMethodRef.EnclosingTypeRef.FullName, name, args, ilinfo.ILMethodRef.GenericArity)
+                              FSharpFindDeclResult.ExternalDecl (assref.Name, externalSym))
+                      | _ -> None
+
+                  | Item.Property (_, ILProp (_, propInfo) :: _) ->
+                      let constructorAndMeth = 
+                          if propInfo.HasGetter then Some (ExternalSymbol.PropertyGet, propInfo.GetterMethod g)
+                          elif propInfo.HasSetter then Some (ExternalSymbol.PropertySet, propInfo.SetterMethod g)
```

Comment:
Yeah I've just changed this in my new PR - Item.Property doesn't specify whether we're looking at a property get or set. It's not a big deal as property getter/setters are always defined next to each other so it's pretty obvious. It would be nice if we knew this in the service, though.